### PR TITLE
MRG: Make update_anat_landmarks() accept fiducials file

### LINKED ISSUE #968: Converting fiducials read via mne.io.read_fiducials() to BIDS anatomical landmarks
MNE stores fiducials in the `mri` (surface RAS) coordinate system; however, BIDS requires MRI voxel coordinates. @alexrockhill has created the helpful function `get_anat_landmarks()` that generates landmarks in the correct coordinate system based on digitization points in an `Info` and a transformation matrix.

However, I now have a file created via the coregistration GUI (which internally calls `mne.io.write_fiducials()`), and there is no easy way to store these `mri` landmarks in my BIDS MRI's `AnatomicalLandmarkCoordinates` field.

I suppose we could modify `get_anat_landmarks()` to accept existing fiducials (or a path to a fiducials file), skip the first bits of processing it currently does with the `Info` and transfomation matrix, and convert the fiducials into the correct coordinate system.

WDYT?

cc @agramfort

### PR DESCRIPTION
**Pull request by hoechenberger**
Fixes #968

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [x] All comments resolved
- [ ] This is not your own PR
- [x] All CIs are happy
- [x] PR title starts with [MRG]
- [x] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/main/doc/whats_new.rst) is updated
- [x] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

File: mne_bids/sidecar_updates.py [hunk 1]
```diff
@@ -152,14 +158,30 @@ def update_anat_landmarks(
...
                   reliable way to generate the landmark coordinates in the
                   required coordinate system.
+
+        .. note:: If ``path-like``, ``fs_subject`` and ``fs_subjects_dir``
+                  must be provided as well.
+
+        .. versionchanged:: 0.10
+           Added support for ``path-like`` input.
+    fs_subject : str
+        The subject identifier used for FreeSurfer. This If ``None``, defaults
```
Comment (hoechenberger) [inline mne_bids/sidecar_updates.py:178]:
```suggestion
        The subject identifier used for FreeSurfer. If ``None``, defaults
```

File: mne_bids/sidecar_updates.py [hunk 2]
```diff
@@ -231,6 +255,17 @@ def update_anat_landmarks(
             f'bids_path. Tried the following filenames: '
             f'{", ".join([p.name for p in tried_paths])}')
 
+    if not isinstance(landmarks, DigMontage):
```
Comment (hoechenberger) [inline mne_bids/sidecar_updates.py:258]:
```suggestion
    if not isinstance(landmarks, DigMontage):  # it's pathlike
```

**Comment by hoechenberger** [comment]
@agramfort Turns out we **didn't** have an issue with zooming … I just used `sample` fiducials on testing-data MRI 🤦

File: mne_bids/sidecar_updates.py [hunk 3]
```diff
@@ -231,6 +255,17 @@ def update_anat_landmarks(
             f'bids_path. Tried the following filenames: '
             f'{", ".join([p.name for p in tried_paths])}')
 
+    if not isinstance(landmarks, DigMontage):  # it's pathlike
+        if fs_subject is None:
+            fs_subject = bids_path.subject
```
Comment (agramfort) [inline mne_bids/sidecar_updates.py:260]: don't do such magic and do what the docstring says. You must provided the fs_subject in this case

File: mne_bids/write.py [hunk 4]
```diff
@@ -580,9 +580,10 @@ def _mri_landmarks_to_mri_voxels(mri_landmarks, t1_mgh):
         The MRI voxel-space landmark data.
     """
     # Get landmarks in voxel space, using the T1 data
-    vox2ras_tkr = t1_mgh.header.get_vox2ras_tkr()
-    ras2vox_tkr = linalg.inv(vox2ras_tkr)
-    vox_landmarks = apply_trans(ras2vox_tkr, mri_landmarks)  # in vox
+    vox2ras_tkr_t = t1_mgh.header.get_vox2ras_tkr()
+    ras_tkr2vox_t = linalg.inv(vox2ras_tkr_t)
+    vox_landmarks = apply_trans(ras_tkr2vox_t, mri_landmarks)
```
Comment (agramfort) [inline mne_bids/write.py:585]: the changes in this function are purely cosmetic. We should avoid this to save review time

File: mne_bids/sidecar_updates.py (hunk 3 above)
Comment (hoechenberger) [reply mne_bids/sidecar_updates.py:260]: I'm actually **doing** what the docstring says, I just copied this exact behavior from ``get_head_mri_trans()`` for consistency:

https://github.com/mne-tools/mne-bids/blob/b26f062a4e1c68fc83ec69df44cd345841a5aed1/mne_bids/read.py#L919-L920

I don't like it either, but I'd say we either do this in both functions or in neither.

Ok to  remove the magic here and deprecate this behavior in `get_head_mri_trans()?

File: mne_bids/write.py (hunk 4 above)
Comment (hoechenberger) [reply mne_bids/write.py:585]: Sorry, I had originally altered the implementation of this function and later reverted to the previous behavior without reverting the cosmetic changes too. Should I revert?

File: mne_bids/read.py [hunk 5]
```diff
@@ -796,8 +796,10 @@ def get_head_mri_trans(bids_path, extra_params=None, t1_bids_path=None,
         session than the MEG. It is even possible to point to a T1 image stored
         in an entirely different BIDS dataset than the MEG data.
     fs_subject : str | None
-        The subject identifier used for FreeSurfer. If ``None``, defaults to
-        the ``subject`` entity in ``bids_path``.
+        The subject identifier used for FreeSurfer.
+
+        .. versionchanged:: 0.10
+           Deprecated defaulting to ``bids_path.subject`` if ``None``.
```
Comment (agramfort) [inline mne_bids/read.py:802]:
```suggestion
           Does not default anymore to ``bids_path.subject`` if ``None``.
```

File: mne_bids/read.py [hunk 6]
```diff
@@ -917,7 +919,11 @@ def get_head_mri_trans(bids_path, extra_params=None, t1_bids_path=None,
     # neuromag RAS coordinate system in order to compare the with MEG landmarks
     # see also: `mne_bids.write.write_anat`
     if fs_subject is None:
+        warn('Passing "fs_subject=None" has been deprecated and will raise '
```
Comment (agramfort) [inline mne_bids/read.py:922]: I see a typo above "We need to convert the to the" -> "We need to convert to the"

same with "compare the with MEG landmarks"

File: mne_bids/read.py [hunk 7]
```diff
@@ -796,8 +796,10 @@ def get_head_mri_trans(bids_path, extra_params=None, t1_bids_path=None,
         session than the MEG. It is even possible to point to a T1 image stored
         in an entirely different BIDS dataset than the MEG data.
     fs_subject : str | None
```
Comment (agramfort) [inline mne_bids/read.py:798]:
```suggestion
    fs_subject : str
    
```

File: mne_bids/read.py (hunk 6 above)
Comment (hoechenberger) [reply mne_bids/read.py:923]: Resolved

**Comment by hoechenberger** [comment]
@agramfort I've addressed all your comments, this should be good to merge once CI comes back green

**Comment by agramfort** [comment]
ok to merge this as is as it already fixes a practical problem but it seems the code only works if the landmarks passed as DigMontage are already in mri voxels. This should not be needed. We allow the same thing as what we allow with write_anat

@hoechenberger if you want we can merge this and please create an issue for the next stuff to do

**Comment by hoechenberger** [comment]
@agramfort 
> ok to merge this as is as it already fixes a practical problem but it seems the code only works if the landmarks passed as DigMontage are already in mri voxels. This should not be needed. We allow the same thing as what we allow with write_anat

Very interesting observation, I just looked at the docstring and implementation of `write_anat()`; it calls `_get_landmarks()`, which only works with coordinates in scanner RAS and voxel space (and raises an exception otherwise).

The docstring suggests to use `get_anat_landmarks()` to produce landmark coordinates in the required coordinate frame (the function returns coords in voxel space); scanner RAS is never mentioned.

So this is at least a documentation bug / shortcoming.

But it also means that `update_anat_landmarks()` should support scanner RAS coordinates as well.

I will look into this.

**Comment by hoechenberger** [comment]
@agramfort I've decided to document the inconsistency in #978 and will go ahead here and merge for now!

**Pull request merged**